### ISSUE TITLE
Add Pokestar sprites?

### ISSUE THREAD

**Issue by Zrp200** [smogon/pokemon-showdown-client#1085]
I know that the Pokestar Pokemon can't be used anywhere outside of custom battles, but being a custom battler myself it's really annoying when your Pokemon is a default image icon.

From what I've seen from the README, the sprite files are not included here. Is there any way the sprites could be added? Since no back sprites exist, my thoughts are that the given sprites should be reversed like the early necrozma sprites were. 

Let me know if there is anything I can do to help with this.

Here are the best sprites I could find (moving!) (all found [here](https://bulbapedia.bulbagarden.net/wiki/Pokéstar_Studios_opponents), but I listed them for ease of access)

Name | Sprite
:-------: | :-------:
ufo1 | ![ufo](https://user-images.githubusercontent.com/7078074/36930874-83389222-1e5f-11e8-8a93-b913a75e4aad.png)
ufo2 | ![ufo2](https://user-images.githubusercontent.com/7078074/36930850-033abd70-1e5f-11e8-98ae-7e00c8b69e24.png)
white door | ![white_door](https://user-images.githubusercontent.com/7078074/36930906-17c705fe-1e60-11e8-8d67-4350c6bb0bbd.png)
black door | ![black_door](https://user-images.githubusercontent.com/7078074/36930946-04fe7564-1e61-11e8-9d96-e58c027e511c.png)
F-00 | ![F-00](https://user-images.githubusercontent.com/7078074/36930916-79857348-1e60-11e8-9f8b-eac76cd704e6.png)
F-002 | ![f-002](https://user-images.githubusercontent.com/7078074/36931054-33b2e316-1e63-11e8-8ea4-44682401574c.png)
Humanoid | ![humanoid](https://user-images.githubusercontent.com/7078074/36931072-7561e9ba-1e63-11e8-8cdb-00740214e102.png)
brycen-man|![brycen-man](https://user-images.githubusercontent.com/7078074/36931085-efd64222-1e63-11e8-80f4-fead493f1088.png)
Transport | ![transport](https://user-images.githubusercontent.com/7078074/36931103-463eefe2-1e64-11e8-907b-bcdf151c2c4d.png)
Monica | ![monica](https://user-images.githubusercontent.com/7078074/36931112-73cb5d4c-1e64-11e8-8589-3240e84afb8a.png)
MT | ![mt](https://user-images.githubusercontent.com/7078074/36931127-c2293cc0-1e64-11e8-9c38-c4a6fa5d0632.png)
MT2 | ![mt2](https://user-images.githubusercontent.com/7078074/36931131-d9f79900-1e64-11e8-824b-b0a596afd82b.png)
Monster | ![monster](https://user-images.githubusercontent.com/7078074/36931182-a16e98b2-1e65-11e8-9fbc-3406a31c5a36.png)
Spirit | ![majin](https://user-images.githubusercontent.com/7078074/36931183-a1859bca-1e65-11e8-980f-08536bcf4caa.png)
black belt | ![spr_5b2_black_belt](https://user-images.githubusercontent.com/7078074/36931184-a19f12f8-1e65-11e8-97c1-c03ecac76e05.png)
Smeargle |![](https://cdn.bulbagarden.net/upload/1/15/Spr_5b2_235_Pok%C3%A9star.png)

Edit: Added Smeargle

**Comment by Zarel** [comment]
We can add them to the website, but not to the repository.

**Comment by Zarel** [comment]
Adding them to the website will involve someone giving me a zip file of these in PS directory format.

i.e. with file/folder names that match what's currently in https://play.pokemonshowdown.com/sprites/

**Comment by Zrp200** [comment]
hmm... how would I title the images?

**Comment by Zrp200** [comment]
well [here](https://github.com/Zarel/Pokemon-Showdown-Client/files/1777841/sprites.zip) are the animated sprite files, as gifs @Zarel. I'm almost certain I screwed up the file naming though, but it shouldn't take too long for me to rename them. I didn't include the pngs because I didn't if I should scrub the animations or not on them.

**Comment by Zrp200** [comment]
[Here](https://github.com/Zarel/Pokemon-Showdown-Client/files/1777895/pokestar.zip) is that zip @Zarel. I'm pretty sure I screwed up the file names; there aren't any other Pokemon that have spaces in their names as far as I know.

**Comment by Zrp200** [comment]
@Zarel here is the [ZIP](https://github.com/Zarel/Pokemon-Showdown-Client/files/1777983/Pokestar.zip).

**Comment by Zrp200** [comment]
@Zarel [Here](https://github.com/Zarel/Pokemon-Showdown-Client/files/1777997/Pokestar.Sprites.zip) is the ZIP.

**Comment by Zarel** [comment]
Images in the `bw` and `bw-back` folder must be 96x96. The bigger ones will need to be cropped.

**Comment by Zrp200** [comment]
so the ones in `bwani` are fine?

**Comment by Zrp200** [comment]
One thing that has been driving me nuts is that I can't find the prop sprites anywhere on the internet except in a [Youtube video](https://www.youtube.com/watch?v=8b8JBJsbdhM), and I don't know how to take them from there. In addition, the cries can also be found there, but I don't know how to get those either.

**Comment by Zarel** [comment]
Yeah, `bwani` has no size limit

**Comment by Zrp200** [comment]
got the grounded sign!
![sign](https://user-images.githubusercontent.com/7078074/36950632-98813220-1fad-11e8-8cc8-fcb5857c9882.png)

**Comment by Zrp200** [comment]
The current ZIP should be fine, as the ones in there are cropped.

**Comment by Zrp200** [comment]
Ok then, since `bwani` is okay, then the correct zip is [here](https://github.com/Zarel/Pokemon-Showdown-Client/files/1778958/Pokestar.Sprites.zip)

**Comment by Zarel** [comment]
We're missing shiny versions of these.

**Comment by Zarel** [comment]
I'm using the non-shiny versions for now.

**Comment by Zrp200** [comment]
I don't know where to find the shiny versions.

**Comment by Zrp200** [comment]
Hmm... Since the prop formes don't have shiny variations, why not have the prop formes as the shiny variations?

**Comment by Zarel** [comment]
do we already have prop formes?

**Comment by Zarel** [comment]
I think the current approach works fine, just don't have shiny versions at all

**Comment by Zrp200** [comment]
The only one I see on-site is UFO-PropU2.

I just made the prop formes anyway, there's only two variations (four pics, for front and back of course):
![groundedpropback](https://user-images.githubusercontent.com/7078074/36952009-0fb93052-1fbf-11e8-97c0-36ac23a263e1.png)
![groundedpropfront](https://user-images.githubusercontent.com/7078074/36952010-0fd3fd10-1fbf-11e8-9f8d-ec8f52fdd9f5.png)
![hangingpropback](https://user-images.githubusercontent.com/7078074/36952011-0ff27894-1fbf-11e8-8ebe-5f1a54db3cb2.png)
![hangingpropfront](https://user-images.githubusercontent.com/7078074/36952012-10103280-1fbf-11e8-93fb-88df54539efe.png)

**Comment by Zrp200** [comment]
The zip I last included has no shiny formes or props in it, so it should work if we don't include these.

**Comment by Zrp200** [comment]
Actually...
![image](https://user-images.githubusercontent.com/7078074/36952082-eaeda8ba-1fbf-11e8-8051-0408ba103463.png)

It seems that there are prop formes programmed in without sprites.

**Comment by Zrp200** [comment]
actually, can these be put in as gen 6 sprites as well?

**Comment by Zrp200** [comment]
that is the same sprites, but I added `xy` and `xyani`

**Comment by Zrp200** [comment]
I realized I misnamed one of the original ones, so ufo-2 doesn't have a sprite up at the moment
[sprites.zip](https://github.com/Zarel/Pokemon-Showdown-Client/files/1779126/sprites.zip)

**Comment by Zrp200** [comment]
Can [this](https://github.com/Zarel/Pokemon-Showdown-Client/files/1779132/sprites.zip)
be reuploaded? I made one naming error. I also added `xy` and `xyani`.

**Comment by Zarel** [comment]
Can you add sprites for the prop formes, too? They'd be `[speciesname]-[formename].png` etc

**Comment by Zarel** [comment]
Also, `xy` is weird; your sprites should be 120x120.

**Comment by Zarel** [comment]
(`xyani` as usual has no dimensions requirements)

**Comment by Zrp200** [comment]
OK.

**Comment by Zrp200** [comment]
I just noticed that a lot of the prop formes are misspelled. For example: `Pokestar UFO-Ropu2` instead of `Pokestar UFO-PropU2`

**Comment by Zarel** [comment]
Can you show me the places they're sisspelled?

**Comment by Zrp200** [comment]
They can be found here: [`aliases.js`](https://github.com/Zarel/Pokemon-Showdown/blob/7bf397fb55df5e6fb6e4075b5824bb9fd42269a0/data/aliases.js)

**Comment by Zarel** [comment]
@Marty-D I'm going to need you to help decide what to name these things...

**Comment by Marty-D** [comment]
From the B2W2 species text file:
```
UFO
Brycen-Man
MT
MT2
Transport
Giant ⑮
Humanoid
Monster
F-00
Spirit
White Door
Black Door
Prop U1
Prop U2
Prop H1
Prop M1
Prop M2
Prop T1
Prop O1
Prop O2
Prop C1
Prop C2
Prop G1
Prop W1
Prop W2
Prop R1
Prop R2
Prop K1
UFO 2
Giant ⑮ 2
F-002
Black Belt
Smeargle 2
```
(⑮ is a special in-game character that doesn't parse properly, in this case it's ♀)
Each prop has exactly the same data as one of the non-props, so there are really only 17 "Pokemon" here.

This is the same order they are listed in https://bulbapedia.bulbagarden.net/wiki/List_of_Pok%C3%A9mon_by_index_number_(Generation_V), except Bulbapedia probably named them after the movie characters or something. I gave this same list to MacChaeger when he insisted on implementing them; if they don't match then I have no idea what he did. I don't particularly care what they're called as long as they don't collide with existing Pokemon/item/egg group names.

**Comment by Zrp200** [comment]
Wait, is there another Giant sprite that I'm missing?

**Comment by Zrp200** [comment]
I don't want to make this too complicated, so I like the current implementation (with errors fixed). However, `Pokestar UFO-2` should be `Pokestar UFO2`, `Pokestar UFO-PropU2` should be `Pokestar UFO2-Prop`, and the Giants should be given the same treatment, just for simplicity's sake.

**Comment by Marty-D** [comment]
Oh, I was counting based on the entries in the server's pokedex file. `pokestarufopropu2` shouldn't be there at all since it's identical to `pokestarufo2`.

The two Giants are identical and also each have their own identical prop, so there are actually only 16 unique ones. Smeargle 2 is the only one without a prop.

**Comment by Zrp200** [comment]
Oh OK. I'll supply an extra giant sprite.

**Comment by Zrp200** [comment]
I want to clear something up. These are the names I'm using for the sprite files:
`pokestarf00` `pokestarf002` `pokestarmonster` `pokestarufo`
`pokestarufo-2` `pokestargiant` `pokestargiant-2` `pokestarsmeargle`
`pokestarspirit` `pokestarblackdoor` `pokestarwhitedoor` `pokestarhumanoid`
`pokestarmt` `pokestarmt2` `pokestarbrycenman` `pokestarblackbelt`

The prop formes have `-prop` tacked on at the end. Does this work?

**Comment by Marty-D** [comment]
You're missing Transport and have too many Giants. Honestly, the whole alternate form thing is dumb since these are only related by "movie appearances" and not in the data:
![thing](https://user-images.githubusercontent.com/1645989/37073404-c0e178e8-21be-11e8-91cf-0367e19bf137.png) etc.
As you can see, none of them truly have alternate forms, so `otherForms` should just not exist here. `otherForms` is for things that are cosmetic changes anyway; why would anyone want to open the teambuilder and see green squares or identical sprites in that section you screenshotted earlier?

**Comment by Zrp200** [comment]
True.

**Comment by Zrp200** [comment]
Do I need an `xy` folder if it's going to be the exact same thing as the stuff in the `bw` folder? The only reason I'm submitting an `xyani` folder at all is because I want the sprites to be animated when in gen 6 graphics.

**Comment by Zarel** [comment]
Yes, you can and should have both.

**Comment by Zrp200** [comment]
OK.

**Comment by Zrp200** [comment]
While I'm working on resizing things, can [the gen 6 animated sprites](https://github.com/Zarel/Pokemon-Showdown-Client/files/1795649/anisprites.zip) at least be uploaded?

**Comment by Zarel** [comment]
Sure.

**Comment by Zrp200** [comment]
Sorry for the extended wait.
[xysprites.zip](https://github.com/Zarel/Pokemon-Showdown-Client/files/1822192/xysprites.zip)

**Comment by Zarel** [comment]
Your 120x120 sprites appear to be resized and anti-aliased. They need to be padded so they remain pixel art.

**Comment by Zarel** [comment]
In other words: When you resize pixel art, it gets blurry. You can't resize it. You have to just add padding around the image.

**Comment by Zrp200** [comment]
Oh

**Comment by Zrp200** [comment]
Got it.

**Comment by Zrp200** [comment]
[sprites.zip](https://github.com/Zarel/Pokemon-Showdown-Client/files/1850339/sprites.zip)
found some more stuff and added it in an `xydex` folder, but this should work; they are all cropped properly to 120x120. I also fixed some sprites I submitted earlier in addition to the new ones.

**Comment by Zrp200** [comment]
@Zarel

**Comment by Zrp200** [comment]
I suppose it isn't urgent. The main part of this is already uploaded. Just don't forget about me, please.

**Comment by Zarel** [comment]
I'll get to it, have just been busy lately

**Comment by Zarel** [comment]
Okay. Some of them are uncropped/padded to 120x120, and some are still stretched.

The stretched ones include: giant, giant2, mt, mt2, ufo, ufo2, all the -prop ones, etc

**Comment by Zarel** [comment]
I noticed you also didn't send regular `bw` versions of those.

**Comment by Zrp200** [comment]
Huh. Can't get it right.... Let me try again.

**Comment by Zrp200** [comment]
just reworked [xydex](https://github.com/Zarel/Pokemon-Showdown-Client/files/1873911/xydex.zip), it should be completely done. If wanted I'll add the other sprites from `xy` when I'm finally done with that, since there are only a few sprites in `xydex`, as the extra pictures I found didn't include all of the Pokestars, rather just a few.

**Comment by Zrp200** [comment]
I know.

**Comment by Zrp200** [comment]
I'm working on them.

**Comment by Zrp200** [comment]
Done. Checked to see if all of them have the same amount of files, caught a missing one, and fixed it. I included everything because I redid a bunch of the sprites. `xydex` is also included again, just for completeness.
[more sprites.zip](https://github.com/Zarel/Pokemon-Showdown-Client/files/1878461/more.sprites.zip)

**Comment by Zarel** [comment]
I guess the blurriness for the high-res sprites is unavoidable... This will have to do.

**Comment by Zarel** [comment]
Tell me if what I uploaded looks good?

**Comment by Zrp200** [comment]
The files I added for `xydex` also need to be copied over to `xydex-shiny`

**Comment by Zrp200** [comment]
Other than that looks pretty good!

**Comment by Zarel** [comment]
Which prop typoes? Can you explain?

**Comment by Zrp200** [comment]
When selecting the prop forms in teambuilder, they are misspelled as "rop"

**Comment by Zrp200** [comment]
And don't work

**Comment by Zrp200** [comment]
![image](https://user-images.githubusercontent.com/7078074/38397005-19880802-38f0-11e8-91d4-0bf9821a05ab.png)
When one clicks on the box where the first arrow I put is, you get this:
![image](https://user-images.githubusercontent.com/7078074/38397045-4f6e348c-38f0-11e8-80b4-57ce349cdabb.png)
![image](https://user-images.githubusercontent.com/7078074/38397064-64c0c8b8-38f0-11e8-8eca-fb81619388e5.png)

**Comment by Zrp200** [comment]
Centered portraits so they aren't cut off in almost all cases. Took you fixing my comment above for me to notice that this needed to be done.
[xydex-changes.zip](https://github.com/Zarel/Pokemon-Showdown-Client/files/3588658/xydex-changes.zip)

**Comment by Zarel** [comment]
What are the prop forms? You don't seem to have sent dex sprites for them.

**Comment by Zarel** [comment]
Hm, I see the problem now. The forme names are very inconsistent. We'll need to fix that.

Marty's chart makes it unclear how the prop sprites come in. Apparently they're not considered formes? Does that chart just not count cosmetic forms?

**Comment by Zrp200** [comment]
Atm, prop formes now display correctly seemingly everywhere but team validation, which thus makes them impossible to use. Validation still uses the misspelling despite "Prop" displaying correctly in import/export.

I'll add prop formes to `xydex` later today, which should cause them to display correctly in the forme picker window.

**Comment by Zarel** [comment]
As far as I can tell, this is done.
